lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -346,12 +371,18 @@ export class UserAgentApplication {
      * @returns {Promise.<AuthResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
      */
     acquireTokenPopup(request: AuthenticationParameters): Promise<AuthResponse> {
-        if (!request) {
-            throw ClientConfigurationError.createEmptyRequestError();
-        }
+        // append GUID to user set state  or set one for the user if null
+        const state = UrlUtils.getRequestState(request && request.state);
 
         return new Promise<AuthResponse>((resolve, reject) => {
-            this.acquireTokenInteractive(Constants.interactionTypePopup, false, request, resolve, reject);
+            if (!request) {
+                throw ClientConfigurationError.createEmptyRequestError();
+            }
+            this.acquireTokenInteractive(Constants.interactionTypePopup, false, request, resolve, reject, state);
+        }).catch((error: AuthError) => {
+            this.cacheStorage.removeItem(TemporaryCacheKeys.INTERACTION_STATUS);
+            this.cacheStorage.resetTempCacheItems(state);
```

Comment:
These two lines could be factored into a helper function.

Reply:
We do not need to clear the Interaction status as we are doing it as a part of cache clearance, hence removed it now.